src/core/api/clock.ts

```diff
@@ -345,18 +352,30 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : { clock$: Observable<IClockTick>, setCurrentTime: (time: number) => void } {
+  let internalSeek: number | null = null;
+  const setCurrentTime = (time: number) => {
+    internalSeek = time;
+    mediaElement.currentTime = time
+  }
+  const clock$ = observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime,
+      });
 
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
       const mediaTimings = getMediaInfos(mediaElement, state);
-      const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
+      const stalledState = getStalledStatus(lastTimings, mediaTimings, options, internalSeek);
+      if (stalledState === null && internalSeek) {
```

Comment:
Same here we check for booleans usually, and that code doesn't have the expected behavior when internalSeek is equal to `0`.

Reply:
Also what if the `reason` changed and we are still stalled but not due to the internal-seek anymore?

Reply:
In my last commit, I edited the `if` condition to be more granular.